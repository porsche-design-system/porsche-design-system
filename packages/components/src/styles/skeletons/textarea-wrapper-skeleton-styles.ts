import {
  extendPseudoWithThemeJssStyle,
  getAfterMinHeight,
  getBaseSkeletonJssStyle,
  getElementBackgroundGradient,
  getHiddenLabelJssStyle,
  getSkeletonElementHeight,
  getSkeletonPropertyNames,
  getThemedPseudoJssStyle,
  LABEL_HEIGHT,
  LABEL_HEIGHT_SPACING,
  LABEL_HEIGHT_WITH_DESCRIPTION,
  TEXTAREA_SKELETON_HEIGHT,
} from './';
import { PDS_SKELETON_CLASS_PREFIX } from '@porsche-design-system/shared';
import { pxToRemWithUnit } from '../common-styles';
import { Styles } from 'jss';

export const getTextareaWrapperSkeletonStyles = (): Styles<'@global'> => {
  const skeletonPropertyNames = getSkeletonPropertyNames('p-textarea-wrapper');

  return {
    '@global': {
      'p-textarea-wrapper': {
        '&:not(.hydrated)': {
          ...extendPseudoWithThemeJssStyle({
            styleFunction: getBaseSkeletonJssStyle,
            pseudosToExtend: ['&::before', '&::after'],
          }),
          minHeight: getSkeletonElementHeight(TEXTAREA_SKELETON_HEIGHT),

          // TODO: use constants for getComponentMeta for "property" class and values
          // hideLabel or no description and no label
          [`&[${skeletonPropertyNames.hideLabel}=true], &.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.hideLabel}, &:not([${skeletonPropertyNames.description}]):not([${skeletonPropertyNames.label}]):not(.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.description}):not(.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.label})`]:
            getHiddenLabelJssStyle(),
          // description no label and no hideLabel
          [`&[${skeletonPropertyNames.description}]:not([${skeletonPropertyNames.label}]):not(.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.label}), &.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.description}:not(.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.label})`]:
            {
              minHeight: getSkeletonElementHeight(TEXTAREA_SKELETON_HEIGHT, false, true),
              '&::before': {
                height: pxToRemWithUnit(LABEL_HEIGHT),
                background: getElementBackgroundGradient(72, 6, true),
              },
              '&::after': {
                top: pxToRemWithUnit(LABEL_HEIGHT),
                minHeight: getAfterMinHeight(LABEL_HEIGHT),
              },
            },
          // description, label and not hideLabel
          [`&[${skeletonPropertyNames.description}][${skeletonPropertyNames.label}]:not([${skeletonPropertyNames.hideLabel}=true]), &.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.description}.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.label}:not(.${PDS_SKELETON_CLASS_PREFIX}${skeletonPropertyNames.hideLabel})`]:
            {
              minHeight: getSkeletonElementHeight(TEXTAREA_SKELETON_HEIGHT, true, true),
              '&::before': {
                height: pxToRemWithUnit(LABEL_HEIGHT_WITH_DESCRIPTION),
                background: getElementBackgroundGradient(LABEL_HEIGHT_WITH_DESCRIPTION, LABEL_HEIGHT_SPACING, true),
              },
              '&::after': {
                top: pxToRemWithUnit(LABEL_HEIGHT_WITH_DESCRIPTION),
                minHeight: getAfterMinHeight(LABEL_HEIGHT_WITH_DESCRIPTION),
              },
            },

          ...getThemedPseudoJssStyle(true),
        },
      },
    },
  };
};
