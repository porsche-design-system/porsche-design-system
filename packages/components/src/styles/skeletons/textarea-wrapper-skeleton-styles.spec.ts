import { getTextareaWrapperSkeletonCss } from './textarea-wrapper-skeleton-styles';

describe('getTextareaWrapperSkeletonCss()', () => {
  it('should return correct css', () => {
    expect(getTextareaWrapperSkeletonCss()).toMatchSnapshot();
  });
});
