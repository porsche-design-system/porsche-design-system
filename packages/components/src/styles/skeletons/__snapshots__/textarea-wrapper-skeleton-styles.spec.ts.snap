// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct css 1`] = `"p-textarea-wrapper:not(.hydrated){color:transparent;display:block;position:relative;min-height:13.75rem}p-textarea-wrapper:not(.hydrated)::before{top:0;left:0;color:#626669;width:8rem;height:1.5rem;content:\\"\\";position:absolute;animation:opacity var(--p-override-skeleton-animation-duration,1.5s) ease-in-out infinite;background:linear-gradient(transparent,transparent 4px,currentColor 4px,currentColor 20px,transparent 20px,transparent 24px);visibility:visible}p-textarea-wrapper:not(.hydrated)::after{top:1.75rem;left:0;color:#626669;width:100%;content:\\"\\";position:absolute;animation:opacity var(--p-override-skeleton-animation-duration,1.5s) ease-in-out infinite;background:currentColor;min-height:calc(100% - 1.75rem);visibility:visible}p-textarea-wrapper:not(.hydrated)[hide-label=true],p-textarea-wrapper:not(.hydrated).PDS-Skeleton--hide-label,p-textarea-wrapper:not(.hydrated):not([description]):not([label]):not(.PDS-Skeleton--description):not(.PDS-Skeleton--label){height:3rem}p-textarea-wrapper:not(.hydrated)[description]:not([label]):not(.PDS-Skeleton--label),p-textarea-wrapper:not(.hydrated).PDS-Skeleton--description:not(.PDS-Skeleton--label){min-height:13.5rem}p-textarea-wrapper:not(.hydrated)[description][label]:not([hide-label=true]),p-textarea-wrapper:not(.hydrated).PDS-Skeleton--description.PDS-Skeleton--label:not(.PDS-Skeleton--label){min-height:15.25rem}p-textarea-wrapper:not(.hydrated)[theme=dark]::before,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--theme-dark::before{color:#b0b1b2}p-textarea-wrapper:not(.hydrated)[theme=dark]::after,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--theme-dark::after{color:#b0b1b2}p-textarea-wrapper:not(.hydrated)[description][label]:not([hide-label=true])::before,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--description.PDS-Skeleton--label:not(.PDS-Skeleton--label)::before{height:3.25rem;background:linear-gradient(transparent,transparent 4px,currentColor 4px,currentColor 20px,transparent 20px,transparent 28px,currentColor 28px,currentColor 40px,transparent 40px,transparent 52px)}p-textarea-wrapper:not(.hydrated)[description][label]:not([hide-label=true])::after,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--description.PDS-Skeleton--label:not(.PDS-Skeleton--label)::after{top:3.25rem;min-height:calc(100% - 3.25rem)}p-textarea-wrapper:not(.hydrated)[description]:not([label]):not(.PDS-Skeleton--label)::before,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--description:not(.PDS-Skeleton--label)::before{height:1.5rem;background:linear-gradient(transparent,transparent 6px,currentColor 6px,currentColor 18px,transparent 18px,transparent 28px,currentColor 28px,currentColor 60px,transparent 60px,transparent 72px)}p-textarea-wrapper:not(.hydrated)[description]:not([label]):not(.PDS-Skeleton--label)::after,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--description:not(.PDS-Skeleton--label)::after{top:1.5rem;min-height:calc(100% - 1.5rem)}p-textarea-wrapper:not(.hydrated)[hide-label=true]::before,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--hide-label::before,p-textarea-wrapper:not(.hydrated):not([description]):not([label]):not(.PDS-Skeleton--description):not(.PDS-Skeleton--label)::before{content:none}p-textarea-wrapper:not(.hydrated)[hide-label=true]::after,p-textarea-wrapper:not(.hydrated).PDS-Skeleton--hide-label::after,p-textarea-wrapper:not(.hydrated):not([description]):not([label]):not(.PDS-Skeleton--description):not(.PDS-Skeleton--label)::after{top:0;min-height:100%}"`;
